app/models/concerns/taxonomix.rb

```diff
@@ -132,6 +132,44 @@ def scope_by_taxable_ids(scope)
         scope.where("#{table_name}.id IN (#{cached_ids.join(',')})")
       end
     end
+
+    def join_user_organizations
+      join_taxonomies(user_organizations, org_join_arel)
+    end
+
+    def join_user_locations
+      join_taxonomies(user_locations, loc_join_arel)
+    end
+
+    def join_taxonomies(taxonomies, tax_arel)
+      class_arel = base_class.arel_table
+      statement = class_arel.join(tax_arel, Arel::Nodes::OuterJoin).on(
+        class_arel[:id].eq(tax_arel[:taxable_id]).
+        and(tax_arel[:taxable_type].eq(base_class.name)).
+        and(tax_arel[:taxonomy_id].in(taxonomies.pluck(:id)))
+      )
+      joins(statement.join_sources)
+    end
+
+    def taxonomy_join_scope
+      distinct.join_user_organizations.join_user_locations
+    end
+
+    def org_join_arel
+      Arel::Table.new(:taxable_taxonomies).alias("tto")
+    end
+
+    def loc_join_arel
+      Arel::Table.new(:taxable_taxonomies).alias("ttl")
+    end
+
+    def user_organizations
+      Array.wrap(Organization.current || Organization.my_organizations.reorder(nil))
+    end
+
+    def user_locations
+      Array.wrap(Location.current || Location.my_locations.reorder(nil))
+    end
```

Comment:
I fixed the child taxonomies issue for the current context but I haven't use the "Taxonomy.expand" as it doesn't quite make sense to me. It will cause duplicate code run for non-admin user with any context. E.g. run Taxonomy.expand -> then run get_taxonomy_ids. This will call "Organization.my_organizations" to get all children and then call the "get_taxonomy_ids" to get all children again.